### PR TITLE
Return code changes

### PR DESCRIPTION
**Pull request by RadhaGulhane13**
Changes in put and del routine: Returning error code in case of failure instead of calling callback

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -6956,11 +6951,12 @@ static void ut_multi_stream_kv_oper(void)
 			rc = m0_be_tx_open_sync(tx);
 			M0_ASSERT(rc == 0);
 
-			M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+			rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
 						 m0_btree_del(tree,
```
Comment (ydb242) [inline btree/btree.c:6955]: Alignment.

File: btree/btree.c [hunk 2]
```diff
@@ -7525,10 +7519,13 @@ static void btree_ut_kv_oper_thread_handler(struct btree_ut_thread_info *ti)
 			cred = M0_BE_TX_CB_CREDIT(0, 0, 0);
 			btree_callback_credit(&cred);
 
-			M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
-						 m0_btree_del(tree, &rec.r_key,
-							      &ut_cb, 0,
-							      &kv_op, tx));
+			rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+						      m0_btree_del(tree,
+						      		   &rec.r_key,
```
Comment (ydb242) [inline btree/btree.c:7524]: tabs and spaces not consistent.
Kindly check that

File: btree/btree.c [hunk 3]
```diff
@@ -6956,11 +6951,12 @@ static void ut_multi_stream_kv_oper(void)
...
 			M0_ASSERT(rc == 0);
 
-			M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
-						 m0_btree_del(tree,
-							      &del_key_in_tree,
-							      &ut_cb, 0,
-							      &kv_op, tx));
+			rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+						      m0_btree_del(tree,
+								   &del_key_in_tree,
```
Comment (ydb242) [inline btree/btree.c:6956]: 80 char limit?

Comment (RadhaGulhane13) [reply btree/btree.c:6956]: I can not change the name of variable, this ut is written by Shashank.

Comment (ydb242) [reply btree/btree.c:6956]: No issues, you can bring the statement after rc  = on the below line starting at same point as rc or maybe with one space.
This way we wont exceed the 80 char limit.

Comment (RadhaGulhane13) [reply btree/btree.c]: ok

**Review by ydb242 — APPROVED** [review]

File: btree/btree.c [hunk 4]
```diff
@@ -8444,17 +8442,17 @@ static void ut_put_update_del_operation(void)
...
-					 m0_btree_del(tree, &rec.r_key,
-						      &ut_cb, 0, &kv_op, tx));
-		if (del_data.flags == M0_BSC_KEY_NOT_FOUND) {
+		int rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+						  m0_btree_del(tree, &rec.r_key,
+						      	       &ut_cb, 0,
+							       &kv_op, tx));
+		if (rc) {
+			M0_ASSERT(rc == M0_ERR(-ENOENT));
 			printf("M0_BSC_KEY_NOT_FOUND ");
```
Comment (NikhilBirgade) [inline btree/btree.c:8451]: The code will assert on line 8450. `printf` statement can be removed

File: btree/btree.c [hunk 5]
```diff
@@ -8407,9 +8402,9 @@ static void ut_put_update_del_operation(void)
 					 m0_btree_update(tree, &rec, &ut_cb, 0,
 							 &kv_op, tx));
 		if (rc) {
```
Comment (NikhilBirgade) [inline btree/btree.c:8404]: `{}` can be removed.

File: btree/btree.c [hunk 6]
```diff
@@ -8444,17 +8439,16 @@ static void ut_put_update_del_operation(void)
...
-		M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
-					 m0_btree_del(tree, &rec.r_key,
-						      &ut_cb, 0, &kv_op, tx));
-		if (del_data.flags == M0_BSC_KEY_NOT_FOUND) {
-			printf("M0_BSC_KEY_NOT_FOUND ");
+		int rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+						  m0_btree_del(tree, &rec.r_key,
+						      	       &ut_cb, 0,
+							       &kv_op, tx));
+		if (rc) {
```
Comment (NikhilBirgade) [inline btree/btree.c:8446]: `{}` can be removed.

File: btree/btree.c [hunk 7]
```diff
@@ -8444,17 +8439,16 @@ static void ut_put_update_del_operation(void)
...
+			M0_ASSERT(rc == M0_ERR(-ENOENT));
 		} else {
```
Comment (NikhilBirgade) [inline btree/btree.c:8448]: `{}` can be removed.

**Review by NikhilBirgade — APPROVED** [review]

**Pull request merged**